clients/client/src/main/java/org/projectnessie/client/NessieClient.java

```diff
@@ -127,6 +161,28 @@ public Builder withConnectionTimeout(int connectionTimeoutMillis) {
     @Override
     @Deprecated
     public NessieClient build() {
+      AuthType auth =
+          authType != null
+              ? authType
+              : (username != null && password != null) ? AuthType.BASIC : null;
+      if (auth != null) {
+        withAuthenticationFromConfig(
+            s -> {
+              switch (s) {
+                case NessieConfigConstants.CONF_NESSIE_AUTH_TYPE:
+                  return auth.name();
+                case NessieConfigConstants.CONF_NESSIE_USERNAME:
+                  return username;
+                case NessieConfigConstants.CONF_NESSIE_PASSWORD:
+                  return password;
+                case NessieConfigConstants.CONF_NESSIE_AWS_REGION:
+                  return null;
```

Comment:
Would it make sense to throw an `IllegalArgumentException` in this case? It does not look like AWS if fully supported in by this builder.

Reply:
It's a new config option, there was no way to set it in the old builder that's eventually going away.

Reply:
noted